libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt

```diff
@@ -1,16 +1,30 @@
 package com.mapbox.navigation.core.fasterroute
 
 import com.mapbox.api.directions.v5.models.DirectionsRoute
+import java.util.concurrent.TimeUnit
 
 /**
  * Interface definition for an observer that get's notified whenever the SDK finds a faster route to the destination.
  */
 interface FasterRouteObserver {
 
+    /**
+     * Overridable constant to change the frequency of the fastest route update.
+     */
+    fun restartAfterMillis(): Long = TimeUnit.MINUTES.toMillis(2)
+
     /**
      * Invoked whenever a faster route is available for user to use.
      *
      * @param fasterRoute reference to route that is faster than the current route.
      */
     fun onFasterRouteAvailable(fasterRoute: DirectionsRoute)
+
+    /**
+     * Invoked whenever a faster route was inspected but the current route is best.
+     *
+     * @param currentRoute reference to the current route.
+     * @param alternativeRoute reference to the alternative checked.
+     */
+    fun onFastestRoute(currentRoute: DirectionsRoute, alternativeRoute: DirectionsRoute)
```

Comment:
Wondering if this could replace `onFasterRouteAvailable` altogether so we can remove one method from the `interface`. It seems is duplicating behavior somehow (also not following the [Interface Segregation Principle](https://en.wikipedia.org/wiki/Interface_segregation_principle) we're adding `// You're on the fastest route` in our examples 😅).

Maybe something around these lines? 

```kotlin
fun onFasterRoute(currentRoute: DirectionsRoute, alternativeRoute: DirectionsRoute, isFastest: Boolean)
```

What do you think?

cc @abhishek1508

Reply:
I think I'm happy with this suggestion. was also thinking that we want the current route provided in `onFasterRouteAvailable`. so this would solve that too.